### DISCUSSION TITLE
Retrieving environment variable values at install time with the UI disabled

### DISCUSSION THREAD

**Discussion by stephen-quinn-gallagher-com** [oleg-shilo/wixsharp#1702] [Q&A]
I am integrating the use of WixSharp generated msi's in our product installer. The msi might need to install a service with a custom account and I would like to make a password available to the msi. 
A possible solution is to use a command line parameter. This works but does make the password available to a snooper so I have been investigating setting an environment variable to the password and running some code to retrieve the password from the environment and setting an installer property.

This works just fine when the the code is invoked in an event handler triggered by the ManagedProject.UILoaded event.

However, when I invoke msiexec and pass the /qn parameter, the environment variable is not to code triggered by the ManagedProject,Load event.

Is there a mechanism which allows me to get this working?

**Comment by oleg-shilo** [comment]
You have multiple event that you can use. All of them are nothing else but custom actions that are wrapped into more convenient C# events. [See here](https://github.com/oleg-shilo/wixsharp/wiki/Managed-Setup-Model#event-driven-custom-actions). 

ManagedProject.**UI**Loaded is an event that it fired when UI is loaded so it's not fired when you deliberately disable UI (`/qn` param) .

Thus you may want to use other events, e.g. `Project.Load`.

**Reply by stephen-quinn-gallagher-com** [reply]
Hi Oleg,
I have tried both the Load and UILoaded events, As you say, the UILoaded is not fired when the UI is disabled. My guess is that when the entire installation is run on the server, its process context is not the same when the Ui is invoked so the environment variable is not available.

Is there any way for a custom action to be invoked before the server is invoked to run the installation? This might allow a property to be set for use by the server
